tests/components/application_credentials/test_init.py

```diff
@@ -0,0 +1,598 @@
+"""Test the Developer Credentials integration."""
+
+from __future__ import annotations
+
+from collections.abc import Callable, Generator
+import logging
+from typing import Any
+from unittest.mock import AsyncMock, Mock, patch
+
+from aiohttp import ClientWebSocketResponse
+import pytest
+
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.components.application_credentials import (
+    DOMAIN,
+    AuthorizationServer,
+    ClientCredential,
+    async_import_client_credential,
+)
+from homeassistant.const import CONF_CLIENT_ID, CONF_CLIENT_SECRET, CONF_DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_entry_oauth2_flow
+from homeassistant.setup import async_setup_component
+
+from tests.common import mock_platform
+
+CLIENT_ID = "some-client-id"
+CLIENT_SECRET = "some-client-secret"
+DEVELOPER_CREDENTIAL = ClientCredential(CLIENT_ID, CLIENT_SECRET)
+ID = "fake_integration_some_client_id"
+AUTHORIZE_URL = "https://example.com/auth"
+TOKEN_URL = "https://example.com/oauth2/v4/token"
+REFRESH_TOKEN = "mock-refresh-token"
+ACCESS_TOKEN = "mock-access-token"
+
+TEST_DOMAIN = "fake_integration"
+
+
+@pytest.fixture
+async def authorization_server() -> AuthorizationServer:
+    """Fixture AuthorizationServer for mock application_credentials integration."""
+    return AuthorizationServer(AUTHORIZE_URL, TOKEN_URL)
+
+
+@pytest.fixture
+async def config_credential() -> ClientCredential | None:
+    """Fixture ClientCredential for mock application_credentials integration."""
+    return None
+
+
+async def setup_application_credentials_integration(
+    hass: HomeAssistant,
+    domain: str,
+    authorization_server: AuthorizationServer,
+    config_credential: ClientCredential | None,
+) -> None:
+    """Set up a fake application_credentials integration."""
+    hass.config.components.add(domain)
+    mock_platform(
+        hass,
+        f"{domain}.application_credentials",
+        Mock(
+            async_get_authorization_server=AsyncMock(return_value=authorization_server),
+        ),
+    )
+    if config_credential:
+        await async_import_client_credential(hass, domain, config_credential)
+
+
+@pytest.fixture
+def disable_setup() -> bool:
+    """Fixture to disable mock credentials integration."""
+    return False
+
+
+@pytest.fixture(autouse=True)
+async def mock_application_credentials_integration(
+    disable_setup: bool,
+    hass: HomeAssistant,
+    authorization_server: AuthorizationServer,
+    config_credential: ClientCredential | None,
+):
+    """Mock a application_credentials integration."""
+    if disable_setup:
```

Comment:
Override the mock_application_credentials_integration fixture instead per test, via pytest.mark.parametrize.

Reply:
Fixed.